docs/02-guides/prevent-excessive-usage-with-rate-limits.mdx

```diff
@@ -0,0 +1,186 @@
+  Requests responses when a limit is exceeded.
+
+### Service Configuration
+
+The rate limiter requires a Redis configuration to function. Upon instantiation,
+it subscribes to a Redis instance and sets up namespaces for different rate
+limits.
+
+To configure the service, you can add the following to your
+`front-commerce.config.ts`:
+
+```ts title="front-commerce.config.ts"
+import { defineConfig } from "@front-commerce/core/config";
+
+export default defineConfig({
+  // ...other configuration options
+  rateLimiter: {
+    redis: {
+      host: process.env.FRONT_COMMERCE_CLOUD_REDIS_SESSIONS_HOST,
+      port: process.env.FRONT_COMMERCE_CLOUD_REDIS_SESSIONS_PORT || 6379,
+      db: process.env.FRONT_COMMERCE_CLOUD_REDIS_SESSIONS_DB || 2,
+    },
+  },
+});
+```
+
+## Implementing Rate Limits
+
+The service provides two main methods for limiting rates:
+
+- `limitHTTPResource(namespace, requestId, options)`: Limits rates for HTTP
+  requests by namespace and request ID.
+- `limitGraphQLResource(fieldPath, requestId, options)`: Limits rates for
+  GraphQL queries based on the resolver field path.
+
+There is also an additional exported method in the `graphql` exports:
+
+- `limitRateByGraphQLResolver(options, resolver)`: Wraps a GraphQL resolver with
+  rate limiting.
+
+### Remix Loader
+
+In the Remix framework, the rate limiter can be employed within a loader to
+control request frequency based on URL path and user IP.
+
+```ts
+import { FrontCommerceApp } from "@front-commerce/remix";
+import type { LoaderFunctionArgs } from "@remix-run/node";
+
+export const loader = async ({ context, request }: LoaderFunctionArgs) => {
+  const app = new FrontCommerceApp(context.frontCommerce);
+
+  await app.services.RateLimiter.limitHTTPResource(
+    new URL(request.url).pathname,
+    app.user.clientIp,
+    {
+      max: 5,
+      duration: "1m",
+    }
+  );
+
+  // Perform loader-specific logic here
+};
+```
+
+### Remix Action
+
+Similarly, for actions in Remix, the rate limiter ensures that excessive
+requests are controlled at the action level, again based on URL path and user
+IP.
+
+```ts
+import { FrontCommerceApp } from "@front-commerce/remix";
+import type { ActionFunctionArgs } from "@remix-run/node";
+
+export const action = async ({ context, request }: ActionFunctionArgs) => {
+  const app = new FrontCommerceApp(context.frontCommerce);
+
+  await app.services.RateLimiter.limitHTTPResource(
+    new URL(request.url).pathname,
+    app.user.clientIp,
+    {
+      max: 5,
+      duration: "1m",
+    }
+  );
+
+  // Perform action-specific logic here
+};
+```
+
+### GraphQL Resolver
+
+For GraphQL resolvers, the rate limiter can be directly applied to individual
+queries or mutations to manage resource consumption effectively.
+
+```ts
+import { createGraphQLRuntime } from "@front-commerce/core/graphql";
+
+export default createGraphQLRuntime({
+  resolvers: {
+    Query: {
+      helloWorld: async (parent, args, context, info) => {
```

Comment:
**nitpick:** most of the documentation use a direct context spread, and doesn't document `info` which IMO is too technical for userland apps.

**suggestion:** adapt the signature (and usages below)

```suggestion
      helloWorld: async (parent, args, { user, services }) => {
```